python/pyarrow/tests/test_dataset.py

```diff
@@ -3295,8 +3295,8 @@ def test_write_dataset_with_scanner(tempdir):
     dataset = ds.dataset(tempdir, partitioning=["b"])
 
     with tempfile.TemporaryDirectory() as tempdir2:
-        ds.write_dataset(dataset.scanner(columns=["b", "c"]), tempdir2,
-                         format='parquet', partitioning=["b"])
+        ds.write_dataset(dataset.scanner(columns=["b", "c"], use_async=True),
```

Comment:
What happens if you pass an sync scanner?

Reply:
Quickly tried it out and you get `ArrowNotImplementedError: Asynchronous scanning is not supported by SyncScanner`, so that seems fine

Reply:
Yes.  I'm hoping most users are not creating scanners directly so this change should be fairly smooth.